### PR TITLE
Service entry address

### PR DESCRIPTION
**Pull request by spikecurtis**
ServiceEntry Address message support.  Updates Pilot to use istio.io/api from https://github.com/istio/api/pull/471

Needs to be updated once the API PR is merged to point `Gopkg.toml` back to main Istio repo/branch.

### CODE REVIEW

**Comment by costinm** [comment]
Please rebase on release-0.8 - master is frozen and behind..

**Comment by costinm** [comment]
(that typically means just 'edit the PR and change the base')

**Comment by spikecurtis** [comment]
@rshriram AFAIK bookinfo examples are not using v1alpha3

**Comment by rshriram** [comment]
They are. See samples/bookinfo/routing

**Comment by spikecurtis** [comment]
Was looking at samples/bookinfo/kube 

No matter, however!  The ServiceEntry example doesn't use addresses.

File: Gopkg.toml [hunk 1]
```diff
@@ -287,7 +287,8 @@ required = [
 
 [[constraint]]
   name = "istio.io/api"
-  branch = "master"
+  branch = "unix-address"
```
Comment (costinm) [inline Gopkg.toml]: I assume this is WIP ?

File: istioctl/cmd/istioctl/convert/testdata/v1alpha3/egress-rule-tcp-wikipedia-cidr.yaml.golden [hunk 2]
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: default
 spec:
   addresses:
-  - 91.198.174.192/27
+  - ip: 91.198.174.192/27
```
Comment (costinm) [inline istioctl/cmd/istioctl/convert/testdata/v1alpha3/egress-rule-tcp-wikipedia-cidr.yaml.golden]: I don't think adding ip: prefix is an option - this kind of names are in wide use.

How about using a prefix for unix or other kind of addresses ( uds:foo )

File: vendor/istio.io/api/networking/v1alpha3/service_entry.proto [hunk 3]
```diff
@@ -59,7 +59,7 @@ option go_package = "istio.io/api/networking/v1alpha3";
 //     metadata:
 //       name: mtls-mongocluster
 //     spec:
-//       host: mymongodb.somedomain
+//       name: mymongodb.somedomain
```
Comment (costinm) [inline vendor/istio.io/api/networking/v1alpha3/service_entry.proto]: Far too late to go back to the old name.

Comment (spikecurtis) [reply vendor/istio.io/api/networking/v1alpha3/service_entry.proto]: Fixed in istio.io/api PR. Will get picked up when I fixup the vendoring in this PR (after the API PR is merged)

**Comment by spikecurtis** [comment]
API changes fixed here are not required.

**Pull request closed without merging**